providers/fips/self_test_data.inc

```diff
@@ -1027,38 +1027,39 @@ static const unsigned char dh_priv[] = {
     0x40, 0xb8, 0xfc, 0xe6
 };
 static const unsigned char dh_pub[] = {
-    0x95, 0xdd, 0x33, 0x8d, 0x29, 0xe5, 0x71, 0x04,
-    0x92, 0xb9, 0x18, 0x31, 0x7b, 0x72, 0xa3, 0x69,
```

Comment:
NOTE for other reviewers: And so forth, with the extant `dh_pub` replaced by a new correct value.  It turns out that the old value did not in fact match the `dh_priv` above, but this did not matter, because it was never used (the CAST only used the host private key in a key agreement with a peer public key, the host public key could have been NULL).